guides/being-a-tech-lead-at-dxw.md

```diff
@@ -0,0 +1,149 @@
+relationship will end.
+
+As an agency, our clients bring us in when there's something they can't do or
+something they don't have the time to do.
+
+#### Working with clients short of expertise
+
+If our client can't do the thing, then you're explicitly there as the expert.
+You're there because you're the one who knows how to build the software they
+need. That means when you tell the client how to build it, they'll listen.
+
+They might be trying to build their own in-house capability; if so they'll be
+looking to you to help them learn. Yes, you're there to deliver working software
+that meets the needs of their users. But you're also there to help them get
+better. When we leave, we need to leave them with working software and an
+in-house team that has the expertise to maintain and develop it.
+
+However, they might not be trying to build their own team. Although software is
+never done, we need to try our best to get there by the time we leave. Their
+service needs to be:
+
+- maintainable with minimal effort
+- reliable, as there's no standing team to respond to issues
+- feature-complete enough that their users can do everything they need without
+  significant frustration
+
+Getting your software into that state is no mean feat, but knowing that's the
+end state can help your prioritisation from the start.
+
+#### Working with clients short of time
+
+If our client can do the thing, but doesn't have the time, then you're no longer
+the expert. You're a specialist, sure, but if they can do what we can, you're
+one of many. Instead of teaching them our best practices, you have to adopt
+theirs. You might get a few chances to work with some individuals who are keen
+to learn, but really you're there to deliver working software. Any capability
+building is a side effect.
+
+The exit plan in these cases is almost always handing over your service to their
+in-house teams. Many organisations in this state will have internal governance
+to ensure that handover will be possible. So, as well as adopting their best
+practices and technology standards, you'll be spending some of your time
+engaging with the technical governance the client has in place. Hopefully you
+can use the things your team is already doing – like architecture decision
+records – to do that engagement. Sometimes you'll need to do extra work.
+Sometimes quite a lot of it.
+
+This extra work can feel really frustrating when you're trying to ship a
+service, but it's really important. Just as we wouldn't take on a service we had
+no hand in that doesn't meet any of our standards, we shouldn't expect our
+clients to. Part of your role is to build relationships with the technology
+folks in the client's organisation, find out which bits of governance we need to
+care about, and make sure your team is in a good position to satisfy them. Often
+those people you build relationships with will be happy to help if you involve
+them early enough!
```

Comment:
Agree. Potentially someting more.

Sometimes the amount of governance and required processes we need to go through takes an unexpected amount of effort. Should we consider something further here that caveats this scenario? I believe will do our best to meet all governance but when that effort starts to risk delivery, we will present those risks along with alternatives to the client.

Reply:
There's a paragraph on this on 109, which maybe covers some of it?

Reply:
The point of governance coming into conflict with delivery is a useful one. Particularly when the drive for governance and the drive for delivery come from different parts of our client organisation. In that situation we often have to do some amount of mediation (normally with a bias towards delivery, because that's why we're getting paid).

Reply:
Adding this paragraph immediately following, which I think covers it:

> Sometimes the amount of governance we need to go through takes an unexpected
> amount of effort, particularly when the drive for governance comes from a
> different place to the drive for delivery. We do our best to meet all the
> governance needs but when that effort starts to risk delivery, we present those
> risks, along with alternatives, to the client. Ultimately, it's up to them what
> risk they're prepared to carry. We often have to do some amount of mediation in
> that case (usually biased towards delivery - it's what we're there for after
> all).

Reply:
LGTM 👍